src/main.rs

```diff
@@ -1,173 +1,96 @@
-    false
-}
-
-async fn get_mappings(client: &Client, cache_dir: &PathBuf, refresh_mappings: bool) -> Result<Vec<ItemMapping>, Box<dyn std::error::Error>> {
-    let mapping_path: PathBuf = cache_dir.join("mappings.json");
-
-    if should_refresh_mapping(&mapping_path, refresh_mappings) {
-        // Refresh mapping file if needed
-        let mapping_body = client.get(MAPPING_URL)
-            .send()
-            .await?
-            .text()
-            .await?;
-        fs::create_dir_all(mapping_path.parent().unwrap()).expect("Unable to create directories");
-        fs::write(mapping_path, &mapping_body).expect("Unable to save mapping data");
-        let results = serde_json::from_str(&mapping_body)?;
-        return Ok(results);
-    }
-    let reader = BufReader::new(File::open(mapping_path)?);
-    let results = serde_json::from_reader(reader)?;
-    Ok(results)
-}
-
-async fn get_prices(client: &Client, cache_dir: &PathBuf, force_prices: bool) -> Result<PriceResults, Box<dyn std::error::Error>> {
-    let prices_file = cache_dir.join("prices.json");
-    if should_refresh_prices(&prices_file, force_prices) {
-        // Re-download the prices
-        let prices_body = client.get(PRICES_LATEST_URL)
-            .send()
-            .await?
-            .text()
-            .await?;
-        fs::create_dir_all(prices_file.parent().unwrap())?;
-        fs::write(prices_file, &prices_body).expect("Unable to save prices data");
-        let results: PriceResults = serde_json::from_str(&prices_body)?;
-        return Ok(results);
-    }
-    let reader = BufReader::new(File::open(prices_file)?);
-    let results = serde_json::from_reader(reader)?;
-    Ok(results)
-}
+use anyhow::Result;
+use log::info;
+use num_format::{SystemLocale, ToFormattedString};
 
-fn get_matching_item_ids(name: &String, mappings: &Vec<ItemMapping>) -> HashSet<u32> {
-    let lower = name.to_lowercase();
-    HashSet::from_iter(mappings
-        .iter()
-        .filter(|&mapping| mapping.name.to_lowercase().contains(&lower))
-        .map(|mapping| mapping.id))
+use crate::{
+    client::{Client, ItemMapping, PriceResult},
+    config::Config,
+};
+
+fn get_matching_items<S: AsRef<str>>(
+    name: S,
+    mappings: &[ItemMapping],
+) -> impl Iterator<Item = &ItemMapping> {
+    let lower = name.as_ref().to_lowercase();
+
+    mappings.iter().filter_map(move |mapping| {
+        mapping
+            .name
+            .to_lowercase()
+            .contains(&lower)
+            .then_some(mapping)
+    })
 }
 
 fn display_price(name: &String, price: &PriceResult) {
     let locale = SystemLocale::default().unwrap();
-    println!("{} -> high: {}, low: {}", name, price.high.unwrap_or(0).to_formatted_string(&locale), price.low.unwrap_or(0).to_formatted_string(&locale));
+    println!(
+        "{} -> high: {}, low: {}",
+        name,
+        price
+            .high
+            .map(|p| p.to_formatted_string(&locale))
+            .unwrap_or("N/A".into()),
```

Comment:
Does `map` here handle the case where `price.high` is none?

Reply:
https://doc.rust-lang.org/std/option/enum.Option.html#method.map

Map operates on Some values, ignores and returns None values. The unwrap_or provides a default value if None